test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -2193,6 +2193,161 @@ Describe "Invoke-WebRequest tests" -Tags "Feature", "RequireAdminOnWindows" {
             $pathologicalRatio | Should -BeGreaterThan 5
         }
     }
+
+    Context 'WebSession persistence tests' {
+        $connWarning = 'WebSession properties changed: new Http connection will be required'
+
+        It 'Connection persistence maintained' {
+            $Uri = Get-WebListenerUrl
+            $null = Invoke-WebRequest -Uri $uri -SessionVariable Session -wv warnings
+            $Warnings | Should -BeNullOrEmpty
+
+            $null = Invoke-WebRequest -Uri $uri -WebSession $Session -wv warnings
+            $Warnings | Should -BeNullOrEmpty
+
+            $null = Invoke-WebRequest -Uri $uri -WebSession $Session -wv warnings
+            $Warnings | Should -BeNullOrEmpty
+
+            $null = Invoke-WebRequest -Uri $uri -WebSession $Session -wv warnings
+            $Warnings | Should -BeNullOrEmpty
+        }
+
+        It 'Connection persistence outperforms non-persistence' {
+            $Uri = Get-WebListenerUrl -Https
```

Comment:
```suggestion
            $uri = Get-WebListenerUrl -Https
```

Reply:
I have addressed the naming issues in the latest commit.

Reply:
I think you can remove the W.I.P label